rs/tests/nested/nns_recovery/BUILD.bazel

```diff
@@ -0,0 +1,63 @@
+load("@rules_rust//rust:defs.bzl", "rust_library")
+load(
+    "//rs/tests:common.bzl",
+    "IC_GATEWAY_RUNTIME_DEPS",
+    "IMPERSONATE_UPSTREAMS_ENV",
+    "IMPERSONATE_UPSTREAMS_RUNTIME_DEPS",
+    "MAINNET_ENV",
+    "MESSAGE_CANISTER_ENV",
+    "MESSAGE_CANISTER_RUNTIME_DEPS",
+)
+load("//rs/tests:system_tests.bzl", "system_test_nns")
+
+package(default_visibility = ["//rs:system-tests-pkg"])
+
+ENV = MAINNET_ENV | IMPERSONATE_UPSTREAMS_ENV | MESSAGE_CANISTER_ENV | {
+    "RECOVERY_GUESTOS_IMG_PATH": "$(rootpath //ic-os/guestos/envs/recovery-dev:update-img.tar.zst)",
+}
+
+RUNTIME_DEPS = IC_GATEWAY_RUNTIME_DEPS + IMPERSONATE_UPSTREAMS_RUNTIME_DEPS + MESSAGE_CANISTER_RUNTIME_DEPS + [
+    "//ic-os/guestos/envs/recovery-dev:update-img.tar.zst",
+    "//rs/tests:recovery/binaries",
+]
+
+DEPENDENCIES = [
+    # Keep sorted.
+    "//rs/recovery",
+    "//rs/tests/consensus/utils",
+    "//rs/tests/driver:ic-system-test-driver",
+    "//rs/tests/nested:nested",
+    "//rs/types/types",
+    "@crate_index//:anyhow",
+    "@crate_index//:futures",
+    "@crate_index//:rand",
+    "@crate_index//:sha2",
+    "@crate_index//:slog",
+]
+
+RUNNER_DEPENDENCIES = [
+    # Keep sorted.
+    "common",
+    "//rs/tests/driver:ic-system-test-driver",
+    "@crate_index//:anyhow",
+]
```

Comment:
Please inline these dependencies into their use-sites. We started avoiding having these kind of variables because it introduces the risk of accidentally introducing unused dependencies. And having an extra unnecessary layer makes it harder to read.